JamfPro/JamfPro/JamfPro.pq

```diff
@@ -271,751 +117,484 @@ UAPIResource = (baseurl as text, relativepath as text, optional query as record,
+        mobile_devices_table = Table.Combine(updated_page_table[response]),
+        expand_hardware = Table.ExpandRecordColumn(mobile_devices_table,
+            "hardware",
+            {
+                "capacityMb",
+                "availableSpaceMb",
+                "usedSpacePercentage",
+                "batteryLevel",
+                "serialNumber",
+                "wifiMacAddress",
+                "bluetoothMacAddress",
+                "modemFirmwareVersion",
+                "model",
+                "modelIdentifier",
+                "modelNumber",
+                "bluetoothLowEnergyCapable",
+                "deviceId"
+            }),
+        expand_user_and_location = Table.ExpandRecordColumn(expand_hardware, 
+            "userAndLocation", 
+            {
+                "username",
+                "realName",
+                "emailAddress",
+                "position",
+                "phoneNumber",
+                "departmentId",
+                "buildingId",
+                "room",
+                "building",
+                "department"
+            }),
+        expand_purchasing = Table.ExpandRecordColumn(expand_user_and_location,
+            "purchasing",
+            {
+                "purchased",
+                "leased",
+                "poNumber",
+                "vendor",
+                "appleCareId",
+                "purchasePrice",
+                "purchasingAccount",
+                "poDate",
+                "warrantyExpiresDate",
+                "leaseExpiresDate",
+                "lifeExpectancy",
+                "purchasingContact"
+            }
+        ),
+        expand_general = Table.ExpandRecordColumn(expand_purchasing,
+            "general",
+            {
+                "udid",
+                "displayName",
+                "assetTag",
+                "siteId",
+                "lastInventoryUpdateDate",
+                "osVersion",
+                "osRapidSecurityResponse",
+                "osBuild",
+                "osSupplementalBuildVersion",
+                "softwareUpdateDeviceId",
+                "ipAddress",
+                "managed",
+                "supervised",
+                "deviceOwnershipType",
+                "enrollmentMethodPrestage",
+                "enrollmentSessionTokenValid",
+                "lastEnrolledDate",
+                "mdmProfileExpirationDate",
+                "timeZone",
+                "declarativeDeviceManagementEnabled",
+                "airPlayPassword",
+                "locales",
+                "languages"
+            })
+    in
+        expand_general;
+
+mobileDevicesApplicationsImplementation = (baseurl as text, pages_table as table) as table =>
+    let
+        sections = List.Transform({

```

Comment:
We could inline the `APPLICATIONS` into the `UAPIResource` function call below, which should get rid of this list transformation thing altogether. Same applies to all of them, of course.

Totally optional. Up to you if readability is helped by the way it is in here.